botocove/cove_host_account.py

```diff
@@ -81,175 +94,229 @@ def get_cove_sessions(self) -> List[CoveSessionInformation]:
-        client: STSClient = self._get_boto3_client("sts", assuming_session)
-        return client
-
-    def _resolve_target_accounts(self, target_ids: Optional[List[str]]) -> Set[str]:
-        accounts_to_ignore = self._gather_ignored_accounts()
-        logger.info(f"Ignoring account IDs: {accounts_to_ignore=}")
-        accounts_to_target = self._gather_target_accounts(target_ids)
-        final_accounts: Set = accounts_to_target - accounts_to_ignore
-        if len(final_accounts) < 1:
-            raise ValueError(
-                "There are no eligible account ids to run decorated func against"
-            )
-        return final_accounts
-
-    def _gather_ignored_accounts(self) -> Set[str]:
-        ignored_accounts = {self.host_account_id}
-
-        if self.provided_ignore_ids:
-            accs, ous = self._get_validated_ids(self.provided_ignore_ids)
-            ignored_accounts.update(accs)
-            if ous:
-                accs_from_ous = self._get_all_accounts_by_organization_units(ous)
-                ignored_accounts.update(accs_from_ous)
-
-        return ignored_accounts
-
-    def _gather_target_accounts(self, targets: Optional[List[str]]) -> Set[str]:
-        if targets:
-            accs, ous = self._get_validated_ids(targets)
-            if ous:
-                accs_from_ous = self._get_all_accounts_by_organization_units(ous)
-                accs.extend(accs_from_ous)
-            return set(accs)
-        else:
-            # No target_ids passed, getting all accounts in org
-            return self._get_active_org_accounts()
 
-    def _get_validated_ids(self, ids: List[str]) -> Tuple[List[str], List[str]]:
+def _host_account_is_in_organization(client: OrganizationsClient) -> bool:
+    try:
+        client.describe_organization()
+        return True
+    except ClientError as error:
+        if _organizations_service_not_in_use(error):
+            return False
+        raise error
 
-        accounts: List[str] = []
-        ous: List[str] = []
 
-        for current_id in ids:
-            if not isinstance(current_id, str):
-                raise TypeError(
-                    f"{current_id} is an incorrect type: all account and ou id's must be strings not {type(current_id)}"  # noqa E501
-                )
-            if re.match(r"^\d{12}$", current_id):
-                accounts.append(current_id)
-                continue
-            if re.match(r"^ou-[0-9a-z]{4,32}-[a-z0-9]{8,32}$", current_id):
-                ous.append(current_id)
-                continue
-            raise ValueError(
-                f"provided id is neither an aws account nor an ou: {current_id}"
-            )
+def _organizations_service_not_in_use(error: ClientError) -> bool:
+    return error.response["Error"]["Code"] == "AWSOrganizationsNotInUseException"
 
-        return accounts, ous
 
-    def _get_all_accounts_by_organization_units(
```

Comment:
I really like the refactor but this probably deserves it's own PR - it's just so much complexity to change all at once.

Reply:
Indeed, there is a lot being changed in this PR and I'm not 100% certain I haven't broken something.

My tests aren't complete, and to be honest in the current framework I'm not sure how to complete them. Moto doesn't have proper support for multiple accounts yet, so currently it's not possible to do a realistic test of the non-org modes.